### PR TITLE
feat：mc sdk & sdk unit test

### PR DESCRIPTION
**Pull request by wzh425**
# Description

mc sdk & sdk unit test

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    /// <summary>
+    /// Gets a value from the dictionary with given key. Returns default value if can not find.
+    /// </summary>
+    /// <param name="dictionary">Dictionary to check and get</param>
+    /// <param name="key">Key to find the value</param>
+    /// <typeparam name="TKey">Type of the key</typeparam>
+    /// <typeparam name="TValue">Type of the value</typeparam>
+    /// <returns>Value if found, default if can not found.</returns>
+    public static TValue GetOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key)
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs:40]: return type is nullable

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    /// <summary>
+    /// Gets a value from the dictionary with given key. Returns default value if can not find.
+    /// </summary>
+    /// <param name="dictionary">Dictionary to check and get</param>
+    /// <param name="key">Key to find the value</param>
+    /// <typeparam name="TKey">Type of the key</typeparam>
+    /// <typeparam name="TValue">Type of the value</typeparam>
+    /// <returns>Value if found, default if can not found.</returns>
+    public static TValue GetOrDefault<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, TKey key)
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs:53]: same as above

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    /// <summary>
+    /// Gets a value from the dictionary with given key. Returns default value if can not find.
+    /// </summary>
+    /// <param name="dictionary">Dictionary to check and get</param>
+    /// <param name="key">Key to find the value</param>
+    /// <typeparam name="TKey">Type of the key</typeparam>
+    /// <typeparam name="TValue">Type of the value</typeparam>
+    /// <returns>Value if found, default if can not found.</returns>
+    public static TValue GetOrDefault<TKey, TValue>(this ConcurrentDictionary<TKey, TValue> dictionary, TKey key)
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs:66]: same as above

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    /// <summary>
+    /// Gets a value from the concurrent dictionary with given key. Returns default value if can not find.
+    /// </summary>
+    /// <param name="dictionary">Concurrent dictionary to check and get</param>
+    /// <param name="key">Key to find the value</param>
+    /// <param name="factory">A factory method used to create the value if not found in the dictionary</param>
+    /// <typeparam name="TKey">Type of the key</typeparam>
+    /// <typeparam name="TValue">Type of the value</typeparam>
+    /// <returns>Value if found, default if can not found.</returns>
+    public static TValue GetOrAdd<TKey, TValue>(this ConcurrentDictionary<TKey, TValue> dictionary, TKey key, Func<TValue> factory)
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Extensions/DictionaryExtensions.cs:114]: where TKey : notnull

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageEntityTypes.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageEntityTypes
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageEntityTypes.cs:6]: Not a good name

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskReceiverTypes.cs [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageTaskReceiverTypes
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskReceiverTypes.cs:6]: Not a good name

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskSelectReceiverTypes.cs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageTaskSelectReceiverTypes
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskSelectReceiverTypes.cs:6]: Not a good name

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskStatuses.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageTaskStatuses
+{
+    WaitSend = 1,
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTaskStatuses.cs:8]: NotPublished、InProgress、Cancel、Published、PublishedFailed、PartialFailed

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTemplateAuditStatuses.cs [hunk 9]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageTemplateAuditStatuses
+{
+    WaitAudit = 1,
+    Adopt,
+    Fail,
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTemplateAuditStatuses.cs:10]: Waited、Approved、Failed

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTemplateStatuses.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum MessageTemplateStatuses
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/MessageTemplateStatuses.cs:6]: MessageTemplateStates

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/ReceiverTypes.cs [hunk 11]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum ReceiverTypes
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/ReceiverTypes.cs:6]: not a good name

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/WebsiteMessageFilterType.cs [hunk 12]
```diff
@@ -0,0 +1,10 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Enum;
+
+public enum WebsiteMessageFilterType
+{
+    MessageTitle = 1,
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Enum/WebsiteMessageFilterType.cs:8]: Title、Content

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Helper/TypeHelper.cs [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+               type == typeof(TimeSpan) ||
+               type == typeof(Guid);
+    }
+
+    public static T GetDefaultValue<T>()
+    {
+        return default;
+    }
+
+    public static object GetDefaultValue(Type type)
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Infrastructure/Helper/TypeHelper.cs:115]: nullable return type

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Model/GetChannelModel.cs [hunk 14]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.BuildingBlocks.BasicAbility.Mc.Model;
+
+public class GetChannelModel : PaginatedOptions
+{
+    public string Filter { get; set; } = string.Empty;
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/Model/GetChannelModel.cs:8]: Format Code

File: src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/_import.cs [hunk 15]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+global using Masa.BuildingBlocks.BasicAbility.Mc.Enum;
```
Comment (zhenlei520) [inline src/BasicAbility/Masa.BuildingBlocks.BasicAbility.Mc/_import.cs:4]: Sort

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**